devlib/module/devfreq.py

```diff
@@ -0,0 +1,199 @@
+#    Copyright 2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from devlib.module import Module
+from devlib.exception import TargetError
+from devlib.utils.misc import memoized
+
+class DevfreqModule(Module):
+
+    name = 'devfreq'
+
+    @staticmethod
+    def probe(target):
+        path = '/sys/class/devfreq/'
```

Comment:
Could be the folder empty?

Maybe we should check there is at least a subfolder.
Probably you could better use your `list_devices` to check that we get a non empty list of subfolders?

Reply:
What do you mean? `probe()` is just checking whether the path exists.

Reply:
I think @derkling  is questioning whether it is enough that the directory exists, or whether `probe()` should check that is also non-empty (i.e. not only that `devfreq` support has been enabled in the kernel, but also that at least one policy is implemented).

We've recently run into a issue like this with `cpufreq`. See https://github.com/ARM-software/devlib/commit/0d63386343e56ff3f20c4a65f71bcd3f281b1ffa

Reply:
Yes, right.

Reply:
IOW: we should not enable the module if there is really nothing to possibly "control"...